### PR TITLE
fix(console): preserve wrapped terminal URLs with OSC 8 links

### PR DESCRIPTION
**Pull request by sbluemin**
## Summary
- Claude Code Operation 실행 환경에 xterm의 OSC 8 하이퍼링크 지원을 명시하여 긴 HTTP URL이 화면 첫 줄에서 잘리지 않도록 수정했습니다. 전역 Shell 및 임의 실행 명령에는 강제하지 않아 리다이렉트 출력의 non-TTY 판정을 보존합니다.
- 명시적인 `FORCE_HYPERLINK` 설정과 기존 HTTP/HTTPS 검증·브라우저 선택 경로를 보존합니다. 줄 조각을 추측해서 합치는 로직은 추가하지 않습니다.
- 기존 Agent/Shell 실행 계약 테스트를 보강했습니다.

## Test Plan
- [x] Console 및 Desktop 빌드 (`pnpm desktop --remote-debugging-port=19347`의 격리 빌드/실행 경로)
- [x] `pnpm --filter @dotobokuri/fleet-console typecheck`
- [x] 관련 Vitest 4개 파일 / 16개 테스트 통과 (`launch-prompt`, `pty`, `launch-env`, `terminal-options`)
- [x] macOS ARM64, Electron 43.1.0 개발 Desktop, Claude Code 2.1.278 / Sonnet 5 실제 응답으로 재현: 전체 377자 URL이 외부/Fleet 브라우저에서 모두 63자로 잘림
- [x] 수정 후 같은 세션·응답 재개: 외부/Fleet 브라우저 요청 모두 377자 원문과 정확히 일치
- [x] 65열/32열 터미널, 첫 줄·중간 줄·마지막 줄, 짧은 Markdown 링크 라벨 검증
- [x] `git diff --check`; 검증 전용 Desktop·Console·수신 서버·브라우저 탭 정리
- [ ] Windows/Linux 실기기 검증 미실시. Windows 외부 열기의 Electron `shell.openExternal` 2,081자 제한은 별도 기존 제약이며 이 PR 범위 밖입니다.

기존 실행 중인 CLI에는 새 환경 변수가 소급 적용되지 않으므로 Operation 재개/재실행이 필요합니다.

### CODE REVIEW

File: runtime/fleet-console/features/execution/host/terminal/launch-env.ts [hunk 1]
```diff
@@ -6,6 +6,9 @@ export function withTerminalCapabilities(env: NodeJS.ProcessEnv): NodeJS.Process
     // xterm.js의 24-bit SGR 렌더링을 알리되, TERM은 널리 호환되는 terminfo 항목을 유지한다.
     COLORTERM: "truecolor",
     TERM: TERMINAL_TERM,
+    // OSC 8은 표시된 줄 조각이 아니라 원래 URL을 운반한다. xterm이 지원하는 능력을 명시해
+    // CLI가 긴 링크를 일반 텍스트로 낮추지 않게 하되, 사용자가 명시적으로 끈 값은 보존한다.
+    FORCE_HYPERLINK: env.FORCE_HYPERLINK ?? "1",
```
[the remark this answers is not in the document]
Comment (sbluemin) [reply runtime/fleet-console/features/execution/host/terminal/launch-env.ts]: FIX — 전역 Shell의 실제 launch env를 리다이렉트된 자식 프로세스에 전달해 supports-hyperlinks 판정을 재현했습니다: 수정 전 tty=false / hyperlinks=true. FORCE_HYPERLINK 기본값을 공통 PTY에서 Claude Code profile 환경으로 이동한 뒤 tty=false / hyperlinks=false로 확인했습니다. 명시적 override와 원래 긴 URL 수정은 유지하며 기존 실행 계약 테스트를 보강했습니다. 빌드·타입 검사·실앱 재검증 후 수정 커밋을 게시하겠습니다.

**Comment by sbluemin** [comment]
@codex The review feedback has been addressed. Please re-review.

## Addressed
- FIX: FORCE_HYPERLINK 기본값을 공통 PTY에서 Claude Code profile 환경으로 한정했습니다. 전역 Shell과 FLEET_TERMINAL_CMD에는 주입하지 않으며 명시적 override를 보존합니다.

## Validation
- 리다이렉트된 자식: tty=false / hyperlinks=true 문제 재현 후 false/false로 해소 확인.
- Console/Desktop 빌드, Console typecheck, 관련 4파일 16테스트, git diff --check 통과.
- 기존 Sonnet 5 세션 재개 후 외부/Fleet Browser 모두 377자 URL 원문 일치. 32열 마지막 줄 클릭도 전체 URL 유지.

## Notes
- 변경은 원래 긴 링크 결함과 재현된 전역 Shell 회귀에 한정했습니다. Windows 외부 URL 플랫폼 한계는 범위 밖으로 유지합니다.

File: runtime/fleet-console/features/execution/host/agent/launch.ts [hunk 2]
```diff
@@ -276,7 +276,13 @@ async function createAgentCliLaunchSpec(options: {
         resolvedModel = toClaudeGatewayModelId(model);
       }
     }
-    const profile = await options.resolveProfile(options.env, options.cwd, {
+    // Claude Code가 원래 URL을 OSC 8로 출력하도록 지원을 알린다. 공통 Shell에 강제하면
+    // 리다이렉트된 명령 출력에도 제어 문자가 섞이므로 Claude 런치에만 적용한다.
+    // 사용자가 명시한 값은 보존한다.
+    const profileEnv = cliId === "claude"
+      ? { ...options.env, FORCE_HYPERLINK: options.env.FORCE_HYPERLINK ?? "1" }
+      : options.env;
```
[the remark this answers is not in the document]
Comment (sbluemin) [reply runtime/fleet-console/features/execution/host/agent/launch.ts]: 확인된 미해결 blocker로 기록하고 자동 머지를 중단합니다. 설치된 Claude Code 2.1.278의 FORCE_HYPERLINK 처리에는 자식 실행 전 제거 경로가 없고, 실제 terminal-link 2.1.1을 captured stdout 자식으로 실행했을 때 FORCE_HYPERLINK=1에서는 OSC 8이 포함되며 값이 없으면 포함되지 않는 것을 확인했습니다. 원래 긴 링크 동작은 실앱에서 고쳐졌지만, 환경 변수 상속의 부작용을 제거했다고 볼 수 없습니다. 터미널 정체성을 가장하거나 Bash 정책을 넓히는 보정은 추가하지 않습니다. 렌더러에 한정된 지원 방식 또는 다른 근본 접근을 정한 후 다시 검증해야 합니다. 현재 head는 미완료 상태로 보존하며 CI green을 머지 근거로 쓰지 않습니다.

[the remark this answers is not in the document]
Comment (sbluemin) [reply runtime/fleet-console/features/execution/host/agent/launch.ts]: RESOLVED by replacing the approach (dac33f419). `FORCE_HYPERLINK` is no longer set anywhere — the launch path is back to canary, and a Claude Bash child in a live Operation reports `printenv FORCE_HYPERLINK` → `unset`. Claude Code 2.1.280 decides hyperlink support only from environment variables, so no renderer-only switch exists; instead the Console terminal now joins hard-wrapped rows and links the whole URL only when the joined text contains a URL recorded verbatim in the session transcript (confirmed server-side without sending transcript content to the browser). Otherwise it yields nothing and the existing plain-URL detection applies.

**Comment by sbluemin** [comment]
@codex The approach has been replaced to resolve the open blocker. Please re-review.

## Addressed
- FIX (blocker: hyperlink forcing inherited by Claude child commands) — reverted the `FORCE_HYPERLINK` launch override entirely (49acb3258); no environment variable is added to any launch.
- Replacement (cd4dbdf5e) — `terminal-wrapped-links.ts` joins rows that reach the right edge with the following indented rows and links the whole URL only when the joined text contains a URL confirmed by the session transcript through `POST /api/v1/agent/sessions/:id/links`, which returns only URLs already present in the submitted text. No match → nothing, so WebLinksAddon keeps its canary behavior.

## Validation
- Console typecheck, Console/Desktop build, changelog check, `git diff --check`.
- Vitest: 5 files / 24 tests including the new transcript-boundary case (fails when the text filter is removed).
- Real Claude Code 2.1.280 output replayed in headless xterm (65/32 cols, Markdown label) and a live isolated Console Operation (~66 and ~49 cols): every row opens the exact 354-char URL; Claude Bash child reports `FORCE_HYPERLINK` unset.

## Notes
- Windows/Linux devices not exercised; the path relies only on xterm cells and transcript strings (no env, shell, OSC 8 passthrough, or isWrapped). Windows `shell.openExternal` length limit remains out of scope.

File: runtime/fleet-console/features/execution/host/agent/transcript-links.ts [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+    });
+  }
+  const urls = [...found.keys()];
+  return urls.length > MAX_URLS ? urls.slice(-MAX_URLS) : urls;
+}
+
+function trimUrl(raw: string): string {
+  let url = raw;
+  for (;;) {
+    const next = url.replace(TRAILING_PUNCTUATION, "");
```
[the remark this answers is not in the document]
Comment (sbluemin) [reply runtime/fleet-console/features/execution/host/agent/transcript-links.ts]: PARTIAL FIX (2fb5fc693). Reproduced for `;`: a wrapped transcript URL ending in `;` was trimmed and the shortened address was linked. The trailing trim now matches the existing plain-URL detection (WebLinksAddon never accepts `:,.!?~` as a URL's last character but does accept `;` and `_`), so `;`/`_` endings are preserved and open verbatim (re-verified on a wrapped 40-column replay; the real Claude captures still link the full 354-char URL on every row, and mismatches still yield no link).

DECLINE for `.`, `,`, `:`, `!`, `?`: a transcript cannot tell sentence punctuation from a URL ending in those characters, and the terminal's own single-line URL detection already drops them on canary. Keeping the same convention for wrapped rows avoids a new behavior difference between one-line and multi-line links.

**Comment by sbluemin** [comment]
@codex The review feedback has been addressed. Please re-review.

## Addressed
- `transcript-links.ts` TRAILING_PUNCTUATION — keep `;`/`_` URL endings, aligned with WebLinksAddon's last-character rule (reproduced truncation for `;`; now exact).

## Validation
- Console typecheck and build; `agent-chat-mode` 10/10.
- Headless replays: wrapped `;`/`_`-ending URLs open exactly; real Claude captures (65/32 cols, Markdown label) still link the full URL on every row; mismatched transcript still yields no link.

## Notes
- DECLINE: trailing `.,:!?` stay trimmed — indistinguishable from sentence punctuation and already dropped by the existing single-line detection.
